api/src/main/java/by/borge/jarl/Plan.java

```diff
@@ -17,11 +17,11 @@ public class Plan {
     /**
      * Evaluate this plan given <code>input</code> and <code>data</code>.
      *
-     * @param input the input value
      * @param data the data document
+     * @param input the input value
      * @return a {@link ResultSet} representing the evaluation result of this Plan
      */
-    public ResultSet eval(Object input, Map<String, ?> data) {
-        return ResultSet.fromResultsList(plan.eval(input, data));
+    public ResultSet eval(Map<String, ?> data, Object input) {
```

Comment:
I actually prefer the previous order, as more eval calls depend on input than data... if we use input and data in that order, we could have overloads like:

```java
public ResultSet eval()
public ResultSet eval(Object input)
public ResultSet eval(Object input, Map<String, ?> data)
```
This order is also what you used in the test data, so I think that's what comes natural. It's also the order used in the plans, i.e. local 0 (input) and 1 (data)

Reply:
True. But then we should probably trickle that through all the way, and change the order of the args to the plans generated by `jarl.parser.make-plan`, and make it all consistent.

Which'll likely involve changing a lot of tests 😅 , but I'm up for it.

Reply:
Oh, I hadn't thought about that.. but yes, if you're up for it, that'd be great! 👍

Reply:
Don't forget about docs too, in case we've documented the other order there

Reply:
Yea, the clj API will be part of the change 👍 .